addons/sale/models/sale.py

```diff
@@ -509,6 +509,7 @@ def _prepare_invoice(self):
             'partner_id': self.partner_invoice_id.id,
             'partner_shipping_id': self.partner_shipping_id.id,
             'fiscal_position_id': self.fiscal_position_id.id or self.partner_invoice_id.property_account_position_id.id,
+            'journal_id': journal.id,  # company comes from the journal
```

Comment:
useless diff

Reply:
The useless part is actually the one that ensures the invoice is created in the correct company (since the company is related on the journal).

Not that useless, I assure you.